hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -185,12 +186,12 @@ mkLspCommand plid cn title args' = do
 allLspCmdIds :: HasPidCache m => IdePlugins -> m [T.Text]
 allLspCmdIds (IdePlugins m) = concat <$> mapM go (Map.toList (pluginCommands <$> m))
   where
-    go (plid, cmds) = mapM (mkLspCmdId plid . commandName) cmds
+    go (plid, cmds) = mapM (mkLspCmdId plid . commandId) cmds
 
-mkLspCmdId :: HasPidCache m => PluginId -> CommandName -> m T.Text
-mkLspCmdId plid cn = do
+mkLspCmdId :: HasPidCache m => PluginId -> CommandId -> m T.Text
+mkLspCmdId plid cid = do
   pid <- T.pack . show <$> getPidCache
-  return $ pid <> ":" <> plid <> ":" <> cn
+  return $ pid <> ":" <> coerce plid <> ":" <> coerce cid
```

Comment:
Is `coerce` the best way to do this? Are we losing clarity? Could we not use a newtype?

Reply:
Yes, the `coerce` is definitely a bit hairy. I'm a bit uncomfortable passing about raw `Texts`s since generally these ids shouldn't be mutated, so I'd like to keep the newtype if possible. I'll replace the `coerce` with a `let` binding instead?